src/components/TextField/TextField.tsx

```diff
@@ -17,19 +18,39 @@ function getRandomDigits() {
 }
 
 export const TextField: React.FC<Props> = ({
+  pattern,
   name,
   value,
   label = name,
   placeholder = `Enter ${label}`,
   required = false,
-  onChange = () => {},
+  onChange = () => { },
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
-
-  // To show errors only if the field was touched (onBlur)
   const [touched, setTouched] = useState(false);
   const hasError = touched && required && !value;
```

Comment:
```suggestion
  const hasError = touched && required && !value.trim();
```
for fix my prev comment looks like was enough to trim value in hasError check